src/main/java/org/hl7/davinci/priorauth/Endpoint.java

```diff
@@ -35,18 +35,18 @@
      * @param requestType   - the RequestType of the request.
      * @return the desired resource if successful and an error message otherwise
      */
-    public static Response read(String resourceType, Map<String, Object> constraintMap, UriInfo uri,
+    public static ResponseEntity<String> read(String resourceType, Map<String, Object> constraintMap, String uri,
             RequestType requestType) {
         logger.info("GET /" + resourceType + ":" + constraintMap.toString() + " fhir+" + requestType.name());
         if (!constraintMap.containsKey("patient") || constraintMap.get("patient") == null) {
             logger.warning("Endpoint::read:patient null");
```

Comment:
Is there a case where the  ! constraintMap.containsKey("patient") returns false and 
constraintMap.get("patient") == null  is true; that is, is the second condition necessary?  Wouldn't the test
 ! constraintMap.containsKey("patient")   be sufficient?

Reply:
Good point. I do not believe there is. I believe the `containsKey` check was added when the database was updated to use general maps. I added the containsKey check on top of the existing `null` check instead of replacing it